cpp/ql/test/query-tests/Security/CWE/CWE-319/UseOfHttp/test.cpp

```diff
@@ -0,0 +1,52 @@
+
+struct host
+{
+	// ...
+};
+
+host gethostbyname(char *str);
+char *strcpy(char *s1, const char *s2);
+char *strcat(char *s1, const char *s2);
+
+void openUrl(char *url)
+{
+	// ...
+
+	host myHost = gethostbyname(url);
+
+	// ...
+}
+
+void doNothing(char *url)
+{
+}
+
+char *urls[] = { "http://example.com" };
+
+void test()
+{
+	openUrl("http://example.com"); // BAD
+	openUrl("https://example.com"); // GOOD (https)
+	openUrl("http://localhost/example"); // GOOD (localhost)
+	openUrl("https://localhost/example"); // GOOD (https, localhost)
+	doNothing("http://example.com"); // GOOD (URL not used)
+	openUrl(urls[0]); // BAD [NOT DETECTED]
```

Comment:
Is this not detected because `urls` is a global variable? Our support for global variables isn't _great_, and I've been looking for a good motivation to better support it for a while.

Reply:
I thought it was because of the array access, but having expanded it into several test cases, the global variable is clearly the issue.

Reply:
Fantastic (well, "fantastic" in the sense that it gives me a reason to improve this case)! I'll create an issue for better flow through global variables.

Reply:
Created https://github.com/github/codeql-c-team/issues/734 for this.

Reply:
Awesome, thanks!